src/credentials/CHIPCertToX509.cpp

```diff
@@ -543,6 +545,41 @@ static CHIP_ERROR DecodeConvertTBSCert(TLVReader & reader, ASN1Writer & writer,
     return err;
 }
 
+/**
+ * Variant of DecodeConvertTBSCert that handles reading a compact-pdc-identity
+ * where only the subject public key is actually encoded. All other values are
+ * populated / written as the well-known values mandated by the specification.
+ *
+ * Note: The reader must be positioned on the EllipticCurvePublicKey element.
+ */
+static CHIP_ERROR DecodeConvertTBSCertCompactIdentity(TLVReader & reader, ASN1Writer & writer, ChipCertificateData & certData)
+{
+    // Decode the public key, everything else is rigid
+    ReturnErrorOnFailure(reader.Expect(kTLVType_ByteString, ContextTag(kTag_EllipticCurvePublicKey)));
+    ReturnErrorOnFailure(reader.Get(certData.mPublicKey));
+
+    // Populate rigid ChipCertificateData fields
+    certData.mSerialNumber = kNetworkIdentitySerialNumberBytes;
+    certData.mSigAlgoOID   = kOID_SigAlgo_ECDSAWithSHA256;
+    InitNetworkIdentitySubject(certData.mIssuerDN);
+    certData.mNotBeforeTime = kNetworkIdentityNotBeforeTime;
+    certData.mNotAfterTime  = kNetworkIdentityNotAfterTime;
+    InitNetworkIdentitySubject(certData.mSubjectDN);
+    certData.mPubKeyAlgoOID  = kOID_PubKeyAlgo_ECPublicKey;
+    certData.mPubKeyCurveOID = kOID_EllipticCurve_prime256v1;
+    certData.mCertFlags.Set(CertFlags::kExtPresent_BasicConstraints);
+    certData.mCertFlags.Set(CertFlags::kExtPresent_KeyUsage);
+    certData.mKeyUsageFlags = kNetworkIdentityKeyUsage;
+    certData.mCertFlags.Set(CertFlags::kExtPresent_ExtendedKeyUsage);
+    certData.mKeyPurposeFlags = kNetworkIdentityKeyPurpose;
+
+    if (!writer.IsNullWriter())
```

Comment:
Shouldn't this be an early return at the top, or are there cases where we want to just get the certData updated without the TBSCert written out?

Reply:
Yes, this happens when loading a PDC cert using `DecodeChipCert()` without the `kGenerateTBSHash` option.